lib/Sema/CSSimplify.cpp

```diff
@@ -2013,6 +2013,16 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
       return getTypeMatchFailure(locator);
   }
 
+  // To contextual type increase the score to avoid ambiguity when solver can
+  // find more than one viable binding different only in representation e.g.
+  //   let _: (@convention(block) () -> Void)? = Bool.random() ? nil : {}
+  // so same representation should be always favored.
+  auto loc = getConstraintLocator(locator);
+  if (loc->findLast<LocatorPathElt::ContextualType>().hasValue() &&
```

Comment:
Nit: `loc->isForContextualType()`

Reply:
Ah, I did this way because `isForContextualType` looks directly at last element, so in cases of optional for example the last path may be `contextual type -> generic argument`

Reply:
Ah sure! `hasValue` is not required though since Optional defines a bool operator.

Reply:
True, just fixed :)